cmd/commands/server/command_server.go

```diff
@@ -158,14 +167,90 @@ func (cmd *Command) Start() (err error) {
 		return err
 	}
 
-	signer := cmd.createSigner(providerID)
+	providerAddress := common.HexToAddress(providerID.Address)
+	// check if node's identity is registered
+	registered, err := identityRegistry.IsRegistered(providerAddress)
+	if err != nil {
+		return err
+	}
 
+	registrationDataProvider := registry.NewRegistrationDataProvider(cmd.keystore)
+	signer := cmd.createSigner(providerID)
 	cmd.proposalAnnouncementStopped.Add(1)
-	go cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+
+	// if not registered - wait indefinitely for identity registration event
+	if !registered {
+		registrationData, err := registrationDataProvider.ProvideRegistrationData(providerAddress)
+		if err != nil {
+			return err
+		}
+
+		registry.PrintRegistrationData(registrationData)
+
+		go func() {
+			killed := cmd.identityRegistrationWaitLoop(providerID, identityRegistry, stopRegistrationWaitLoop)
+			if killed {
+				cmd.proposalAnnouncementStopped.Done()
+				return
+			}
+			cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+		}()
+	} else {
+		go cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+	}
+
+	err = cmd.registerTequilAPI(identityRegistry, providerID, registrationDataProvider)
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
+func (cmd *Command) registerTequilAPI(statusProvider registry.IdentityRegistry, providerID identity.Identity, registrationDataProvider registry.RegistrationDataProvider) error {
+	registry.AddRegistrationEndpoint(cmd.router, registrationDataProvider, statusProvider, &providerID)
+
+	err := cmd.httpAPIServer.StartServing()
+	if err != nil {
+		return err
+	}
+
+	address, err := cmd.httpAPIServer.Address()
+	if err != nil {
+		return err
+	}
+
+	port, err := cmd.httpAPIServer.Port()
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Api started on: %s:%d", address, port)
+	return nil
+}
+
+func (cmd *Command) identityRegistrationWaitLoop(providerID identity.Identity, identityRegistry registry.IdentityRegistry, stopLoop chan int) bool {
```

Comment:
As I mentioned earlier - it could be separate discovery component which takes care of proposal registration, checking own identity registration is part of that.